cni/network/network.go

```diff
@@ -941,77 +941,85 @@ func (plugin *NetPlugin) Delete(args *cniSkel.CmdArgs) error {
 		}
 	}
 
-	// Initialize values from network config.
-	networkID, err = plugin.getNetworkName(args.Netns, nil, nwCfg)
-	if err != nil {
-		log.Printf("[cni-net] Failed to extract network name from network config. error: %v", err)
-		// If error is not found error, then we ignore it, to comply with CNI SPEC.
-		if !network.IsNetworkNotFoundError(err) {
+	// Loop through all the networks that are created for the given Netns. In case of multi-nic scenario ( currently supported
+	// scenario is dual-nic ), single container may have endpoints created in multiple networks. As all the endpoints are
+	// deleted, getNetworkName will return error of the type NetworkNotFoundError which will result in nil error as compliance
+	// with CNI SPEC as mentioned below.
+
+	numEndpointsToDelete := plugin.nm.GetNumEndpointsInNetNs(args.Netns)
```

Comment:
lets call this api only for multitenancy and keep  default value of `numEndpointsToDelete` to 1, otherwise this would regress a scenario where endpoint not created by azure cni but azure-ipam allocated ip for container id

Reply:
solved